src/java/org/apache/ivy/plugins/resolver/AbstractPatternsBasedResolver.java

```diff
@@ -91,6 +91,9 @@ protected ResolvedResource findResourceUsingPatterns(ModuleRevisionId moduleRevi
         Set<String> foundRevisions = new HashSet<>();
         boolean dynamic = getSettings().getVersionMatcher().isDynamic(moduleRevision);
         for (String pattern : patternList) {
+            if (Thread.currentThread().isInterrupted()) {
+                return null;
```

Comment:
@apupier , I don't think this is the best place to check the thread interruption, but I don't have a better suggestion right now, so unless someone objects, this change is probably fine. However, instead of returning null, can you please throw a (runtime) exception instead.

Reply:
the specific exception in this case is in theory https://docs.oracle.com/javase/7/docs/api/java/lang/InterruptedException.html which is a checked Exception.
Do you have another runtime exception in mind? Or you want to create a specific Runtime Exception? I'm wondering what would be the effect in this case to Thread in which it has been launched. Will it finished also?

Reply:
@apupier I originally intended it to be `InterruptedException` but given that it's a checked exception you would end up changing the method signature of this method which then causes issues with callers of this API. Hence I suggested a runtime exception.  Returning null  could have other (side) effects than what you intend here - i.e. to abort the execution, so throwing the exception would be the right thing.

I might have some time tonight, I'll see if this check can be moved to a better location or handle in a better way.